### PR TITLE
Add quick fix for S4621 ('no-duplicate-in-composite')

### LINKED ISSUE #3027: Add quick fix for S4621: Union and intersection types should not be defined with duplicated elements
SonarJS rule.

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #3027

### CODE REVIEW

File: eslint-bridge/src/rules/no-duplicate-in-composite.ts [hunk 1]
```diff
@@ -54,6 +55,7 @@ export const rule: Rule.RuleModule = {
 
         groupedTypes.forEach(duplicates => {
           if (duplicates.length > 1) {
+            const suggest = getSuggestions(compositeType, duplicates.slice(1), context);
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/no-duplicate-in-composite.ts:58]: minor but I would move slicing logic inside of the function

File: eslint-bridge/src/rules/no-duplicate-in-composite.ts [hunk 2]
```diff
@@ -66,10 +68,38 @@ export const rule: Rule.RuleModule = {
...
+): Rule.SuggestionReportDescriptor[] {
+  const uniqueTypes = composite.types.filter(tpe => !duplicates.includes(tpe));
+  const uniqueTexts = uniqueTypes.map(tpe =>
+    context.getSourceCode().getText(tpe as unknown as estree.Node),
+  );
+  const compositeNode = composite as unknown as estree.Node;
+  const firstToken = context.getSourceCode().getFirstToken(compositeNode);
+  const lastToken = context.getSourceCode().getLastToken(compositeNode);
+  let [prefix, suffix] = ['', ''];
+  if (firstToken?.value === '(' && lastToken?.value === ')') {
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/no-duplicate-in-composite.ts:93]: I don't see test for this

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
TBH I am not big fan of such solution based on full text replacement (and not on removal of duplicates), but indeed it's pretty straight-forward.

File: eslint-bridge/src/rules/no-duplicate-in-composite.ts (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/src/rules/no-duplicate-in-composite.ts:93]: https://github.com/SonarSource/SonarJS/blob/777bc114e0fca7a251be1fcafe01d1dc20a5e5d3/eslint-bridge/tests/rules/no-duplicate-in-composite.test.ts#L164-L198

Comment (vilchik-elena) [reply eslint-bridge/src/rules/no-duplicate-in-composite.ts:94]: hm... here first and last parentheses are not a pair, that's why I didn't get that it covers the condition
Then for `(number | string) & (number | string) & Foo` we gonna get `number | string & Foo` which is wrong.

**Review by vilchik-elena — CHANGES_REQUESTED** [review]

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**